keras_cv/models/generative/stable_diffusion/text_encoder.py

```diff
@@ -0,0 +1,125 @@
+import tensorflow as tf
+from tensorflow import keras
+import numpy as np
+from tensorflow.experimental import numpy as tfnp
+
+
+class TextEncoder(keras.Model):
+    def __init__(self, max_length, name=None):
+        input_tokens = keras.layers.Input(
+            shape=(max_length,), dtype="int32", name="input_tokens"
+        )
+        token_positions = keras.layers.Input(
+            shape=(max_length,), dtype="int32", name="token_positions"
+        )
+
+        input_dim, output_dim = 49408, 768
+
+        x = CLIPEmbedding(input_dim, output_dim, max_length)(
+            [input_tokens, token_positions]
+        )
+        for _ in range(12):
+            x = CLIPEncoderLayer()(x)
```

Comment:
A general question -- how much of those text-related encoder do we want to declare as private API?

Reply:
In my opinion all text APIs should be __internal__ - CLIP should eventually be in KerasNLP so we should make it clear this won’t live here long term.

Reply:
Yeah, and a key point here is that the Stable Diffusion *cannot be repurposed for anything else*. Our only goal is to match the numerics needed to load the open-source weights (the checkpoint cost ~800k to produce so we won't retrain). We won't use standard components or make components from the model reusable. Even the conv2d we use is non-standard (`PaddedConv2D`).

Reply:
Will this really be the general approach to "fundation models" in keras-cv?

It seems really an outlier for the library modularity manifesto.

We have started a thread at https://github.com/keras-team/keras-cv/issues/488#issuecomment-1154998812

Unfortunately, between internal team meetings and public discussions we never manage to stay on the same page. As community we always live with an information asymmetry / debt